src/com/company/GameManager.java

```diff
@@ -19,10 +19,20 @@
     private static final String FORMATTED_PROVIDE_NAME = "%s %d %s \n";
     private static final String FORMATED_WELCOME = "%s %s %s %s \n";
 
-    public static void play() {
+    public void play() {
 
-        boolean shouldPlayAgain = true;
         final Player players[] = new Player[NUMBER_OF_PLAYERS];
+        initPlayers(players);
+        playGame(players);
+        ioManager.showMessage(END_OF_THE_GAME);
+    }
+
+    public static int getRandomPlayerIndex(final int playerCount) {
+        final Random random = new Random();
+        return random.nextInt(playerCount);
+    }
+
+    public void initPlayers(Player players[]) {
```

Comment:
Lepiej byłoby gdyby ta metoda zwracała tablicę a nie ją przyjmowała.
Jeśli stworzysz ją w środku to będziesz pewna jaki ma rozmiar, przyjmując ją z zewnątrz możesz mieć tylko nadzieję, że będzie wielkości `NUMBER_OF_PLAYERS`.

Reply:
Przeniosłam ten kawałek do środka playGame (tworzenie tablicy i wywołanie inicjowania graczy). Ale teraz jest bardzo mało w play, czy to ma sens? Coś mi się tu nie podoba, ale nie wiem co ;)

Reply:
ale samej logiki nie rozumiem, przecież ja tworze tablicę, chwile przed wywołaniem metody, to dlaczego mam nie być pewna tego jaki ma rozmiar?

Reply:
Zrobiłaś coś zupełnie innego niż prosiłem :P 
Funkcja initPlayers miała zwracać tablice graczy a zamiast tego jej wywołanie zostało przeniesione, dlaczego?

1) Proszę przenieś spowrotem, inicjowanie tej tablicy powinno być przed `playGame()`
2) jak już zauważyłaś tworzysz tablicę przed wywołaniem metody, to znaczy ze jak zmienisz tę linijkę z jej tworzeniem to twoja funkcja przestanie działać bo w jej wnętrzu jest zaszyte sporo logiki opierającej się na tym rozmiarze, np:
```
for (int i = 0; i < NUMBER_OF_PLAYERS; i++) {
[...]
players[i] = new Player(name, sign[i]);
```

Tablicę `sign` tworzysz w środku ale `players` już na zewnątrz. Co jak zmienisz rozmiar jednej z nich? Powyższa linijka się wykrzaczy lub zadziała źle ale dopiero w trakcie działania programu.
Łatwo to będzie przeoczyć jak już zapomnisz że ta funkcja jest zależna od tamtej linijki.
Jesli będą w jednym miejscu to będzie to odrazu widoczne.
Pozatym skąd inny programista który chciałby wykorzystać tę metodę miałby wiedzieć, że tablica przekazywana do niej musi mieć konkretny rozmiar?

Reply:
Powiedzmy że wróciłam do poprzedniej wersji, czyli tworze tablicę, wywołuję initPlayers i nastepnie PlayGame. Jeśli tablica ma być tworzona w initPlayers to i tak muszę ją zadeklarować w play (wyżej), czyli najpierw deklaruję tablicę, potem w podmetodzie ją tworzę po to żeby ją przekazać jako wynik.

Reply:
A może zamiast korzystać z tej stałej powinnaś ją przekazać jako argument?
Wtedy ta metoda byłaby jak najbardfziej niezależna od reszty programu i byłaby testowalna :)

Reply:
nie rozumiem, zamiast używać stałej przekazać ją jako argument? Na pierwszy rzut oka można tym zabiegiem tylko ograniczyć widoczność stałej. Do przedyskutowania.

Reply:
ok, porozmawiamy o tym na lekcji